test/e2e/network_segmentation_default_network_annotation.go

```diff
@@ -103,4 +103,84 @@ var _ = Describe("Network Segmentation: Default network multus annotation", func
+				"e2e-framework":           f.BaseName,
+				RequiredUDNNamespaceLabel: "",
+			})
+			Expect(err).NotTo(HaveOccurred(), "Should create namespace for test")
+			f.Namespace = namespace
+
+			udnClient, err := udnclientset.NewForConfig(f.ClientConfig())
+			Expect(err).NotTo(HaveOccurred(), "Should create UDN client")
+
+			// Create a UserDefinedNetwork for the test
+			udn := &udnv1.UserDefinedNetwork{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-network",
+					Namespace: f.Namespace.Name,
+				},
+				Spec: udnv1.UserDefinedNetworkSpec{
+					Topology: udnv1.NetworkTopologyLayer2,
+					Layer2: &udnv1.Layer2Config{
+						Role: udnv1.NetworkRolePrimary,
+						Subnets: filterDualStackCIDRs(f.ClientSet, []udnv1.CIDR{
+							udnv1.CIDR("103.0.0.0/16"),
+							udnv1.CIDR("2014:100:200::0/60"),
+						}),
+					},
+				},
+			}
+
+			By("Creating a UserDefinedNetwork")
+			udn, err = udnClient.K8sV1().UserDefinedNetworks(f.Namespace.Name).Create(context.TODO(), udn, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Should create UserDefinedNetwork")
+			Eventually(userDefinedNetworkReadyFunc(f.DynamicClient, udn.Namespace, udn.Name), 5*time.Second, time.Second).Should(Succeed())
+
+			By("Creating a pod without the default-network annotation")
+			podWithoutAnnotation := e2epod.NewAgnhostPod(f.Namespace.Name, "pod-without-annotation", nil, nil, nil)
+			podWithoutAnnotation.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+			podWithoutAnnotation = e2epod.NewPodClient(f).CreateSync(context.TODO(), podWithoutAnnotation)
+
+			By("Creating a pod with the default-network annotation")
+			ips := []string{"103.0.0.3/16", "2014:100:200::3/60"}
+			ipsJSON, err := json.Marshal(ips)
+			Expect(err).NotTo(HaveOccurred(), "Should marshal IPs for annotation")
+
+			podWithAnnotation := e2epod.NewAgnhostPod(f.Namespace.Name, "pod-with-annotation", nil, nil, nil)
+			podWithAnnotation.Annotations = map[string]string{
+				"v1.multus-cni.io/default-network": fmt.Sprintf(`[{"name":"default", "namespace":"ovn-kubernetes", "mac":%q, "ips": %s}]`, "02:A1:B2:C3:D4:E5", string(ipsJSON)),
+			}
+			podWithAnnotation.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+			podWithAnnotation = e2epod.NewPodClient(f).CreateSync(context.TODO(), podWithAnnotation)
+
+			By("Attempting to add the default-network annotation to the pod without annotation")
+			podWithoutAnnotation.Annotations["v1.multus-cni.io/default-network"] = `[{"name":"default", "namespace":"ovn-kubernetes", "mac":"02:A1:B2:C3:D4:E5", "ips": ["103.0.0.3/16"]}]`
+
+			_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Update(context.TODO(), podWithoutAnnotation, metav1.UpdateOptions{})
+			Expect(err).To(HaveOccurred(), "Should fail to add default-network annotation to existing pod")
+			Expect(err.Error()).To(ContainSubstring("The 'v1.multus-cni.io/default-network' annotation cannot be changed after the pod was created"))
```

Comment:
```suggestion
			Expect(err).To(MatchError(ContainSubstring("The 'v1.multus-cni.io/default-network' annotation cannot be changed after the pod was created")), "Should fail to add default-network annotation to existing pod")
```

The match error matcher accepts other matchers as inputs; plus if you match error, you expect an error, which renders expecting the error redundant. 

It's opinionated, I know.